From 12a4457d222c36d7468489fd2ba36fe4a2163d11 Mon Sep 17 00:00:00 2001
From: ran shalit <rshalit@habana.ai>
Date: Tue, 27 Jun 2023 17:04:11 +0300
Subject: [PATCH] [SW-149942] efuse: add init as part of general flow

efuse init was recently removed, as part of a patch
which removes tpm enable:
[SW-142278] h6: tpm: remove tpm_en efuse bit check

efuse init was done in code in the first attempt to read
efuse, hence - since no one reads anything from efuse
(as part of the above patch), there is no init and any other
later project (preboot, mgmt etc) fails in efuse read.

This patch fix this issue by forcing efuse init as part
of general initialization in main.

Change-Id: I308b7cf69a825bf85850a02dc02331be59bf0537
Signed-off-by: ran shalit <rshalit@habana.ai>
(cherry picked from commit da0f484970e66f0cbbc9fec873318662844e25cd)
---

diff --git a/app/ppboot/ppboot.c b/app/ppboot/ppboot.c
index 9d6504d..030d8ab 100644
--- a/app/ppboot/ppboot.c
+++ b/app/ppboot/ppboot.c
@@ -19,6 +19,7 @@
 #include "pll.h"
 #include <boot_if.h>
 #include <pci.h>
+#include <efuse.h>
 #include <hl.h>
 
 struct device_ca53_cfg ca53_cfg;
@@ -340,6 +341,7 @@
 
 	preboot_sync_btl();
 	hl_uart_init();
+	preboot_fuse_init();
 	spi_flash_prot_early();
 	ppboot_security_init();
 	hl_pcie_init();
